### PR TITLE
Add documentation about using custom OpenAI models

### PR DESCRIPTION
**Pull request by tm-kn**

### CODE REVIEW

File: docs/ai-backends.md [hunk 1]
```diff
@@ -107,3 +107,46 @@ WAGTAIL_AI = {
     }
 }
 ```
+
+#### Using a custom OpenAI or OpenAI-compatible model
+
+llm supports adding custom OpenAI models. This may be necessary if:
+
+- You want to use a model that's not supported by llm package yet.
```
[the remark this answers is not in the document]
Comment (tm-kn) [reply docs/ai-backends.md:115]: I do wonder if we want to monospace the library name. We don't write `Wagtail`, `Windows`, `Django` etc.

[the remark this answers is not in the document]
Comment (zerolab) [reply docs/ai-backends.md:115]: I don't mind either way, as long as we are consistent throughout

**Comment by tm-kn** [comment]
@zerolab Thanks for your help. I've updated the documentation now and made it more consistent. Let me know how you feel about it.

The "LLM" library's name is all uppercase (https://llm.datasette.io/en/stable/) so I've just started using that instead and I've added the double quotes to make sure people don't think it means the acronym for large language models.

File: docs/ai-backends.md [hunk 2]
```diff
@@ -107,3 +112,47 @@ WAGTAIL_AI = {
...
+
+The "LLM" library supports adding custom OpenAI models. This may be necessary if:
+
+- You want to use a model that's not supported by the "LLM" library yet.
+- You want to use a proxy for OpenAI requests.
+
+You can find the "LLM" library specific instructions at: https://llm.datasette.io/en/stable/other-models.html#adding-more-openai-models.
+
+1. Find the "LLM" library's directory. You can set a custom one with the. You can
+   set a custom one with the
```
Comment (zerolab) [inline docs/ai-backends.md:126]:
```suggestion
1. Find the "LLM" library's directory. You can set a custom one with the
```

File: docs/ai-backends.md [hunk 3]
```diff
@@ -107,3 +112,47 @@ WAGTAIL_AI = {
...
+   [`LLM_USER_PATH`](https://llm.datasette.io/en/stable/setup.html#setting-a-custom-directory-location)
+   setting. You can then confirm the path with the following shell command:
+   `dirname "$(llm logs path)"`.
```
Comment (zerolab) [inline docs/ai-backends.md:129]:
```suggestion
   setting. To confirm the path, you can use the following shell command:
   `dirname "$(llm logs path)"`.
```

File: docs/ai-backends.md [hunk 4]
```diff
@@ -107,3 +112,47 @@ WAGTAIL_AI = {
...
+   For example to set up a proxy:
+   ```yaml
+    - model_id: customgateway-gpt-3.5-turbo
+      model_name: gpt-3.5-turbo
+      api_base: "https://yourcustomproxy.example.com/"
+      headers:
+        apikey: your-api-key
+   ```
+3. Set the `MODEL_ID` in the Wagtail AI settings in your Django project
+   settings to the `model_id` you set in the yaml file.
```
Comment (zerolab) [inline docs/ai-backends.md:141]:
```suggestion
   settings file to the `model_id` you added in `extra-openai-models.yaml`.
```

**Review by zerolab — CHANGES_REQUESTED** [review]
A few minor tweaks and this is good to merge.
Thank you @tm-kn

**Comment by tm-kn** [comment]
Thanks for your help @dan!

**Comment by tm-kn** [comment]
Thanks for your help @zerolab!

**Pull request merged**